presences/src/main/java/fr/openent/presences/service/impl/DefaultAbsenceService.java

```diff
@@ -64,8 +64,13 @@ public void get(String structureId, String startDate, String endDate,
         JsonArray params = new JsonArray();
         String query = "SELECT * FROM " + Presences.dbSchema + ".absence";
 
-        query += " WHERE structure_id = ? AND start_date > ? AND end_date < ? OR ? > start_date ";
+        query += " WHERE structure_id = ? AND ((start_date > ? AND end_date < ?) ";
+        query += " OR (start_date < ? AND end_date > ?) ";
+        query += " OR (start_date < ? AND end_date > ?))";
```

Comment:
WHERE structure_id = ? AND start_date <= ? AND end_date >= ? ...
params.add(structureId).add(structureId).add(endAt).add(startAt) ...

ça me parait plus simple, non ? ;)

Reply:
je comprends pas comment ce code regle le pb du milti etab

Reply:
son code ou mon code ?
dans les 2 cas, le problème était le OR, en SQL:
A & B | C
ça fait A & B = AB
donc AB | C
=> sauf que à droite du | , on veut B aussi, or dans l'ancienne écriture, bhé y'a pas B ^^